File: Medium/Python/1343. Number of Sub-arrays of Size K and Average Greater than or Equal to Threshold.py
class Solution(object):
    def numOfSubarrays(self, arr, k, threshold):
        count = 0
        this_sum = 0
        this_avg = 0

        for i in range(len(arr)):
            this_sum += arr[i]
            if k - 1 <= i:
                this_avg = this_sum / k
                if threshold <= this_avg:
                    count += 1
                this_sum -= arr[i - (k - 1)]

        return count

File: Medium/Java/1343. Number of Sub-arrays of Size K and Average Greater than or Equal to Threshold.java
class Solution {
    public int numOfSubarrays(int[] arr, int k, int threshold) {
        int count = 0;
        int sum = 0;
        int avg = 0;

        for(int i = 0; i < arr.length; i++){
            sum += arr[i];
            if(k - 1 <= i){
                avg = sum / k;
                if(threshold <= avg)
                    count++;
                sum -= arr[i - (k - 1)];
            }
        }

        return count;
    }
}
